sequence_processing_pipeline/Pipeline.py

```diff
@@ -12,11 +12,79 @@
 import sample_sheet
 import pandas as pd
 from collections import defaultdict
+from datetime import datetime
+import pytz
+from xml.etree import ElementTree as ET
 
 
 logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
 
 
+class InstrumentUtils():
+    types = {'A': 'NovaSeq 6000', 'D': 'HiSeq 2500', 'FS': 'iSeq',
+             'K': 'HiSeq 4000', 'LH': 'NovaSeq X Plus', 'M': 'MiSeq',
+             'MN': 'MiniSeq',
+             # SN – RapidRun which is HiSeq 2500
+             'SN': 'RapidRun'}
+
+    @staticmethod
+    def get_instrument_type(run_id):
+        tmp = run_id.split('_')
+
+        # remove digits from substring to get instrument code.
+        code = ''.join([i for i in tmp[1] if not i.isdigit()])
+        try:
+            return InstrumentUtils.types[code]
+        except KeyError:
+            raise ValueError(f"Instrument code '{code}' is of unknown type")
+
+    @staticmethod
+    def get_instrument_id(run_id):
+        tmp = run_id.split('_')
+
+        # remove letters from substring to get instrument id.
+        return ''.join([i for i in tmp[1] if i.isdigit()])
+
+    @staticmethod
+    def get_date(run_id):
+        ds = run_id.split('_')[0]
+
+        if len(ds) == 6:
+            year = int(ds[0:2]) + 2000
+            month = int(ds[2:4])
+            date = int(ds[4:6])
+        elif len(ds) == 8:
+            year = int(ds[0:4])
+            month = int(ds[4:6])
+            date = int(ds[6:8])
+        else:
+            raise ValueError("Couldn't parse datestamp: '%s'" % ds)
+
+        if month < 1 or month > 12:
+            raise ValueError("Invalid month: '%s'" % month)
+
+        if date < 1 or date > 31:
+            raise ValueError("Invalid date: '%s'" % date)
+
+        return datetime(year, month,
```

Comment:
don't understand why is important to use the timezone, can you clarify? If is important we might need to know which sequencing runs were done in Boulder.

Reply:
I can go either way on this as well. Right now, there isn't a specific need to extract the date stamp but since I was here I felt it was worthwhile to add. The timezone was really just a reminder to myself that the user may need a date stamp w/timezone. I agree if some of these were done in Boulder then we can't assume they're all Pacific/Los Angeles.

My assumption is datestamp would default to 12:00:00AM, in which case assuming the date could change if they assume UTC time vs San Diego, or whatevers. It's a good question though I will try it out later to see what happens.

I'm all for removing the time-zone and just noting that in the docstring, since we don't have a user for this method and hence no requirements.